src/shared/components/lazyMobXTable/LazyMobXTable.tsx

```diff
@@ -208,29 +213,33 @@ class LazyMobXTableStore<T> {
     @computed public get downloadData()
     {
         const tableDownloadData:string[][] = [];
-
+        
         // add header (including hidden columns)
         tableDownloadData[0] = [];
         this.columns.forEach((column:Column<T>) => {
-            tableDownloadData[0].push(column.name);
+            if (column.toDownload) {
+                tableDownloadData[0].push(column.name);
+            }
         });
 
         // add rows (including hidden columns)
         this.dataStore.sortedData.forEach((rowData:T) => {
-            const rowDownloadData:string[] = [];
-
-            this.columns.forEach((column:Column<T>) => {
-                if (column.download) {
-                    rowDownloadData.push(column.download(rowData));
+            if (rowData instanceof Array) {
+                for (let sample of rowData) {
+                    const rowDownloadData:string[] = [];
+                    this.columns.forEach((column:Column<T>) => {
+                        if (column.toDownload) {
+                            if (column.download) {
+                                rowDownloadData.push(column.download(rowData, sample));
+                            } else {
+                                rowDownloadData.push("");
+                            }
+                        }
+                    });
+                    tableDownloadData.push(rowDownloadData);
                 }
-                else {
-                    rowDownloadData.push("");
-                }
-            });
-
-            tableDownloadData.push(rowDownloadData);
+            }
```

Comment:
decision: use extra "else"  here for when `rowData` is not an Array (which is indeed not always the case).